pkgs/tools/package-management/nix/default.nix

```diff
@@ -75,6 +76,11 @@ common =
             patchelf --set-rpath $out/lib:${stdenv.cc.cc.lib}/lib $out/lib/libboost_thread.so.*
           ''}
         '' +
+        # Unpack the Rust crates.
+        ''
+          tar xvf ${crates} -C nix-rust/
```

Comment:
You probably already know this, but: since `crates` can be `null`, there should be a check here.